modules/mauth-signer-akka-http/src/main/scala/com/mdsol/mauth/http/TraceHttpRequest.scala

```diff
@@ -0,0 +1,20 @@
+package com.mdsol.mauth.http
+
+import akka.http.scaladsl.model.headers.RawHeader
+import akka.http.scaladsl.model.{HttpHeader, HttpRequest}
+import brave.Span
+
+import scala.collection.immutable.Seq
+
+final class TraceHttpRequest(val request: HttpRequest) {
+  def withTraceHeaders[T](childSpan: Span): TraceHttpRequest = {
```

Comment:
arguably, this logic should be in the constructor or apply method on companion object. Otherwise it's possible to construct a TraceHttpRequest object that is not properly initialised.

Reply:
I believe it should still be possible to make untraced calls, but could provide a companion objects such as 
```
object TraceHttpRequest {
  def apply(request: HttpRequest): TraceHttpRequest = new TraceHttpRequest(request)

  def apply(request: HttpRequest, childSpan: Span): TraceHttpRequest = new TraceHttpRequest(request).withTraceHeaders(childSpan)
}```

Reply:
It shouldn't be possible in TracedClient though? We should force tracing there.